extension/EdFi.Ods.Extensions.LMSX/LMS-Harmonizer/1090-lms-assignment-submissions.sql

```diff
@@ -41,9 +42,52 @@ BEGIN
 		ON EDFISTUDENT.Id = lmsUser.EdFiStudentId
     WHERE lmsSubmission.SourceSystem = @SourceSystem
 
+	-- Here we are building missing submissions when they are not present
+	-- We do this to standardize what is shown in the visualizations
+	ALTER TABLE
+		#ALL_SUBMISSIONS
+	ALTER COLUMN
+		AssignmentSubmissionIdentifier
+	int NULL;
+
+
+	INSERT INTO #ALL_SUBMISSIONS
+	SELECT
+		edfisectionassociation.SectionIdentifier
+			+ '#' + lmsxassignment.AssignmentIdentifier
+			+ '#' + lmsstudent.SourceSystemIdentifier
+			as SourceSystemIdentifier,
```

Comment:
I really like reading this syntax, what do you think? I'm not attached to it, so if you prefer the original syntax  that's fine.

```suggestion
		FORMATMESSAGE(
		    '%s#%s#%s,
    		edfisectionassociation.SectionIdentifier,
			lmsxassignment.AssignmentIdentifier,
			lmsstudent.SourceSystemIdentifier,
		) as SourceSystemIdentifier,
```